### ISSUE TITLE
Slash and backslash + modifier in NeoVim seems inverted using Kitty

### ISSUE THREAD

**Issue by metal3d** [kovidgoyal/kitty#8046]
**Describe the bug**

Note that I'm using an AZERTY keyboard.

Using NeoVim + LazyVim, the shortcut should`CTRL+/` opens a terminal. I tried with WezTerm, Gnome-Terminal, and Xterm => it's OK

But, with Kitty, I have to press `CTRL+\` instead. So it's not so important, I only had to remind that I need this shortcut instead.
Pressing `CTRL+/` (that is Ctrl + Shift + / on AZERTY keyboard) is not OK with Kitty.

In editing files, the `\` and `/` keys are OK. And the shortcut `/` to grep text is OK too. It seems that the only problem appends with the `CTRL` modifier in "normal mode".

I checked the keymap on pressing `CTRL+/` (which is CTRL+Shift+/ on AZERTY):

```
shift+ctrl+num_lock+: PRESS 
CSI 58 : 47 : 46 ; 134 u
Shifted key: / Alternate key: . 
```

And for `CTRL+\` that I have to use instead on NeoVim:

```
ctrl+num_lock+_ PRESS 
CSI 95 :  : 56 ; 133 u
Alternate key: 8 
```

**Environment details**
```
kitty 0.37.0 created by Kovid Goyal
Linux patrice-laptop-home 6.11.7-300.fc41.x86_64 #1 SMP PREEMPT_DYNAMIC Fri Nov  8 19:23:10 UTC 2024 x86_64
S
Kernel 6.11.7-300.fc41.x86_64 on an x86_64 (/dev/tty)

Running under: Wayland (GNOME Shell 47.1) missing: layer_shell
OpenGL: '4.6 (Core Profile) Mesa 24.2.6' Detected version: 4.6
Frozen: False
Fonts:
  medium: JetBrainsMonoNF-Regular: /home/metal3d/.local/share/fonts/jetbrainsmono-nerd-font/JetBrainsMonoNerdFont-Regular.ttf:0
          Features: ()
    bold: JetBrainsMonoNF-SemiBold: /home/metal3d/.local/share/fonts/jetbrainsmono-nerd-font/JetBrainsMonoNerdFont-SemiBold.ttf:0
          Features: ()
  italic: JetBrainsMonoNF-Italic: /home/metal3d/.local/share/fonts/jetbrainsmono-nerd-font/JetBrainsMonoNerdFont-Italic.ttf:0
          Features: ()
      bi: JetBrainsMonoNF-SemiBoldItalic: /home/metal3d/.local/share/fonts/jetbrainsmono-nerd-font/JetBrainsMonoNerdFont-SemiBoldItalic.ttf:0
          Features: ()
Paths:
  kitty: /usr/bin/kitty
  base dir: /usr/lib64/kitty
  extensions dir: /usr/lib64/kitty/kitty
  system shell: /bin/bash
Loaded config files:
  /home/metal3d/.config/kitty/kitty.conf

Config options different from defaults:
dynamic_background_opacity True
enable_audio_bell          False
font_family                JetBrainsMono Nerd Font
font_size                  9.0
sync_to_monitor            False
tab_bar_style              powerline
Added shortcuts:
	ctrl+alt+p →  kitten zoom_toggle.py
	ctrl+alt+s →  goto_layout stack
	ctrl+alt+t →  goto_layout tall
	ctrl+down →  resize_window shorter 1
	ctrl+left →  resize_window narrower
	ctrl+right →  resize_window wider
	ctrl+up →  resize_window taller 1
Changed shortcuts:
	kitty_mod+down →  next_window
	kitty_mod+enter →  new_window_with_cwd
	kitty_mod+n →  new_os_window_with_cwd
	kitty_mod+t →  new_tab_with_cwd
	kitty_mod+up →  prev_window
Colors:
	active_tab_background      #99c1f1   
	active_tab_foreground      #fafafa   
	background                 #242424   
	color0                     #242424   
	color1                     #ff7b63   
	color10                    #26a269   
	color11                    #cd9309   
	color12                    #3584e4   
	color13                    #613583   
	color14                    #4d7b73   
	color15                    #5a6169   
	color2                     #8ff0a4   
	color3                     #f8e45c   
	color4                     #99c1f1   
	color5                     #dc8add   
	color6                     #7fbeb3   
	color7                     #fafafa   
	color8                     #1e1e1e   
	color9                     #c01c28   
	cursor                     #b7bcb9   
	foreground                 #fafafa   
	inactive_tab_background    #f6f5f4   
	inactive_tab_foreground    #5e5c64   
	selection_background       #5e5c64   
	selection_foreground       #f6f5f4   

Important environment variables seen by the kitty process:
	PATH                                /usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin
	LANG                                fr_FR.UTF-8
	EDITOR                              /usr/bin/nano
	SHELL                               /bin/bash
	DISPLAY                             :0
	WAYLAND_DISPLAY                     wayland-0
	USER                                metal3d
	XDG_DATA_DIRS                       /home/metal3d/.local/share/flatpak/exports/share:/var/lib/flatpak/exports/share:/usr/local/share/:/usr/share/
	XDG_RUNTIME_DIR                     /run/user/1000
	XDG_CURRENT_DESKTOP                 GNOME
	XDG_MENU_PREFIX                     gnome-
	XDG_SESSION_CLASS                   user
	XDG_SESSION_DESKTOP                 gnome
	XDG_SESSION_TYPE                    wayland

```

Note: Weird it sees "nano" as EDITOR while it's actually `nvim` that I set in my `.bashrc.d/` : 

```
$ echo $EDITOR
nvim
```

(Thanks a lot for this terminal !)

**Comment by kovidgoyal** [comment]
You should report the issue to neovim, this is almost certainly neovim
interpreting kitty keyboard protocol events incorrectly.

If you think kitty is producing incorrect key events, check it by running:

kitty --debug-input kitten show-key -m kitty

and posting both the log from --debug-input and the output of the show
key kitten when you press the problematic keys.

**Comment by metal3d** [comment]
As I said, the problem is only with kitty-terminal + neovim. The same machine with xterm, WezTerm, gnome-terminal, and so on... does what is expected.

This is the log when I press `CTRL+\` on kitty:
```
[412,553] Press xkb_keycode: 0x1d clean_sym: Control_L composed_sym: Control_L mods: numlock glfw_key: 57442 (LEFT_CONTROL) xkb_key: 65507 (Control_L)
[412,553] on_key_input: glfw key: 0xe062 native_code: 0xffe3 action: PRESS mods: ctrl+numlock text: '' state: 0 sent encoded key to child: ^[ [ 5 7 4 4 2 ; 1 3 3 u 
[413,001] Press xkb_keycode: 0x9 clean_sym: underscore composed_sym: underscore mods: ctrl+numlock glfw_key: 95 (_) xkb_key: 95 (underscore) alternate_key: 56 (8)
[413,001] on_key_input: glfw key: 0x5f native_code: 0x5f action: PRESS mods: ctrl+numlock text: '' state: 0 sent encoded key to child: ^[ [ 9 5 : : 5 6 ; 1 3 3 u 
[413,049] Release xkb_keycode: 0x9 clean_sym: underscore mods: ctrl+numlock glfw_key: 95 (_) xkb_key: 95 (underscore) alternate_key: 56 (8)
[413,049] on_key_input: glfw key: 0x5f native_code: 0x5f action: RELEASE mods: ctrl+numlock text: '' state: 0 sent encoded key to child: ^[ [ 9 5 : : 5 6 ; 1 3 3 : 3 u 
[414,467] Release xkb_keycode: 0x1d clean_sym: Control_L mods: ctrl+numlock glfw_key: 57442 (LEFT_CONTROL) xkb_key: 65507 (Control_L)
[414,467] on_key_input: glfw key: 0xe062 native_code: 0xffe3 action: RELEASE mods: numlock text: '' state: 0 sent encoded key to child: ^[ [ 5 7 4 4 2 ; 1 2 9 : 3 u 
```

And this with `wev`:

```
[14:     wl_keyboard] key: serial: 28313; time: 15083863; key: 37; state: 1 (pressed)
                      sym: Control_L    (65507), utf8: ''
[14:     wl_keyboard] modifiers: serial: 28314; group: 0
                      depressed: 00000004: Control 
                      latched: 00000000
                      locked: 00000010: Mod2 
[14:     wl_keyboard] key: serial: 28315; time: 15084632; key: 17; state: 1 (pressed)
                      sym: underscore   (95), utf8: ''
[14:     wl_keyboard] key: serial: 28316; time: 15085146; key: 17; state: 0 (released)
                      sym: underscore   (95), utf8: ''
[14:     wl_keyboard] key: serial: 28317; time: 15085830; key: 17; state: 1 (pressed)
                      sym: underscore   (95), utf8: ''
[14:     wl_keyboard] key: serial: 28318; time: 15085888; key: 17; state: 0 (released)
                      sym: underscore   (95), utf8: ''
[14:     wl_keyboard] key: serial: 28319; time: 15086110; key: 37; state: 0 (released)
                      sym: Control_L    (65507), utf8: ''
[14:     wl_keyboard] modifiers: serial: 28320; group: 0
                      depressed: 00000000
                      latched: 00000000
                      locked: 00000010: Mod2 

```

It seems that one says that the keycode is 95 and the others says 17. Am I wrong? (it's possible 😉 )

**Comment by metal3d** [comment]
Sorry, for CTRL+Shift+/

with kitty:

```
[4,975] Press xkb_keycode: 0x1d clean_sym: Control_L composed_sym: Control_L mods: numlock glfw_key: 57442 (LEFT_CONTROL) xkb_key: 65507 (Control_L)
[4,975] on_key_input: glfw key: 0xe062 native_code: 0xffe3 action: PRESS mods: ctrl+numlock text: '' state: 0 sent encoded key to child: ^[ [ 5 7 4 4 2 ; 1 3 3 u 
[5,515] Press xkb_keycode: 0x36 clean_sym: Shift_R composed_sym: Shift_R mods: ctrl+numlock glfw_key: 57447 (RIGHT_SHIFT) xkb_key: 65506 (Shift_R)
[5,515] on_key_input: glfw key: 0xe067 native_code: 0xffe2 action: PRESS mods: ctrl+shift+numlock text: '' state: 0 [5,515] 
text-input: updating cursor position: left=2 top=69 width=7 height=16
sent encoded key to child: ^[ [ 5 7 4 4 7 ; 1 3 4 u 
[5,516] text-input: done event: serial: 12 current_commit_serial: 12
[6,142] Press xkb_keycode: 0x34 clean_sym: colon composed_sym: slash mods: ctrl+shift+numlock glfw_key: 58 (:) xkb_key: 58 (colon) shifted_key: 47 (/) alternate_key: 46 (.)
[6,142] on_key_input: glfw key: 0x3a native_code: 0x3a action: PRESS mods: ctrl+shift+numlock text: '' state: 0 [6,142] 
text-input: updating cursor position: left=2 top=117 width=7 height=16
sent encoded key to child: ^[ [ 5 8 : 4 7 : 4 6 ; 1 3 4 u 
[6,143] text-input: done event: serial: 13 current_commit_serial: 13
[6,319] Release xkb_keycode: 0x34 clean_sym: colon mods: ctrl+shift+numlock glfw_key: 58 (:) xkb_key: 58 (colon) shifted_key: 47 (/) alternate_key: 46 (.)
[6,320] on_key_input: glfw key: 0x3a native_code: 0x3a action: RELEASE mods: ctrl+shift+numlock text: '' state: 0 [6,320] 
text-input: updating cursor position: left=2 top=181 width=7 height=16
sent encoded key to child: ^[ [ 5 8 : 4 7 : 4 6 ; 1 3 4 : 3 u 
[6,320] text-input: done event: serial: 14 current_commit_serial: 14
[6,537] Release xkb_keycode: 0x36 clean_sym: Shift_R mods: ctrl+shift+numlock glfw_key: 57447 (RIGHT_SHIFT) xkb_key: 65506 (Shift_R)
[6,537] on_key_input: glfw key: 0xe067 native_code: 0xffe2 action: RELEASE mods: ctrl+numlock text: '' state: 0 [6,537] 
text-input: updating cursor position: left=2 top=245 width=7 height=16
sent encoded key to child: ^[ [ 5 7 4 4 7 ; 1 3 3 : 3 u 
[6,537] text-input: done event: serial: 15 current_commit_serial: 15
[6,583] Release xkb_keycode: 0x1d clean_sym: Control_L mods: ctrl+numlock glfw_key: 57442 (LEFT_CONTROL) xkb_key: 65507 (Control_L)
[6,583] on_key_input: glfw key: 0xe062 native_code: 0xffe3 action: RELEASE mods: numlock text: '' state: 0 [6,583] 
text-input: updating cursor position: left=2 top=293 width=7 height=16
sent encoded key to child: ^[ [ 5 7 4 4 2 ; 1 2 9 : 3 u 

```

with wev:

```
[14:     wl_keyboard] key: serial: 29366; time: 15343711; key: 37; state: 1 (pressed)
                      sym: Control_L    (65507), utf8: ''
[14:     wl_keyboard] modifiers: serial: 29367; group: 0
                      depressed: 00000004: Control 
                      latched: 00000000
                      locked: 00000010: Mod2 
[14:     wl_keyboard] key: serial: 29368; time: 15344061; key: 62; state: 1 (pressed)
                      sym: Shift_R      (65506), utf8: ''
[14:     wl_keyboard] modifiers: serial: 29369; group: 0
                      depressed: 00000005: Shift Control 
                      latched: 00000000
                      locked: 00000010: Mod2 
[14:     wl_keyboard] key: serial: 29370; time: 15344385; key: 60; state: 1 (pressed)
                      sym: slash        (47), utf8: ''
[14:     wl_keyboard] key: serial: 29371; time: 15344511; key: 60; state: 0 (released)
                      sym: slash        (47), utf8: ''
[14:     wl_keyboard] key: serial: 29372; time: 15344881; key: 62; state: 0 (released)
                      sym: Shift_R      (65506), utf8: ''
[14:     wl_keyboard] modifiers: serial: 29373; group: 0
                      depressed: 00000004: Control 
                      latched: 00000000
                      locked: 00000010: Mod2 
[14:     wl_keyboard] key: serial: 29374; time: 15344899; key: 37; state: 0 (released)
                      sym: Control_L    (65507), utf8: ''
[14:     wl_keyboard] modifiers: serial: 29375; group: 0
                      depressed: 00000000
                      latched: 00000000
                      locked: 00000010: Mod2 

```

Both seems to send good combination. So, what could be the problem ?

**Comment by kovidgoyal** [comment]
On Wed, Nov 13, 2024 at 04:43:40AM -0800, Patrice Ferlet wrote:
> As I said, the problem is only with kitty-terminal + neovim. The same machine with xterm, WezTerm, gnome-terminal, and so on... does what is expected.

That doesnt mean anything. None of those terminals implement the kitty
keyboard protocol with the exception of wezterm behind a config flag.

I told you what command you need to run if you want further help,
run it and post the log BOTH from --debug-input and the actual keys
displayed by the show-key kitten. I don't know why you seem to think you
know better than I do what is needed here.

wev's key codes have no connection to terminal key codes.

**Comment by metal3d** [comment]
OK, so the log I gave you in the initial comment + my answers comes from your commands output. I never said that I know better than you or that you're wrong. I only try to understand the problem.

There's no reason to be aggressive and disparage my comments. Especially since I follow the commands you give me and paste the results as you indicate.

So, potentially, your request isn't clear.

**Comment by metal3d** [comment]
So, one more time, but now the ENTIRE log:

```
[0.045] Ignoring unknown config key: open_url_modifiers
[0,071] Loading new XKB keymaps
[0,077] Modifier indices alt: 0x3 super: 0x6 hyper: 0xffffffff meta: 0xffffffff numlock: 0x4 shift: 0x0 capslock: 0x1 control: 0x2
[0,150] 
text-input: updating cursor position: left=2 top=5 width=7 height=16
[0,173] on_focus_change: window id: 0x1 focused: 1
[0,173] 
text-input: updating IME focus state, ime_focused: 0 ev->focused: 1
[0,173] 
text-input: updating IME focus state, ime_focused: 0 ev->focused: 1
[0,173] text-input: enter event
[0,173] on_focus_change: window id: 0x1 focused: 1
[0,173] 
text-input: updating IME focus state, ime_focused: 1 ev->focused: 1
[0,182] Mouse cursor entered window: 1 at 609,964844x19,085938
[0,182] Pushed key encoding flags to: 31
[0,185] text-input: done event: serial: 5 current_commit_serial: 5
[0,190] Got notification server capabilities: frozenset({'actions', 'sound', 'persistence', 'body', 'body-markup', 'icon-static'})
[0,739] Move x: 622,4 y: 186,4 grabbed: 0
[0,746] Move x: 622,1 y: 175,8 grabbed: 0
[0,753] Move x: 621,3 y: 158,0 grabbed: 0
[0,760] Move x: 620,8 y: 132,6 grabbed: 0
[0,767] Move x: 619,3 y: 106,8 grabbed: 0
[0,774] Move x: 617,4 y: 85,3 grabbed: 0
[0,781] Move x: 615,1 y: 66,8 grabbed: 0
[0,789] Move x: 612,2 y: 49,9 grabbed: 0
[0,795] Move x: 609,0 y: 34,9 grabbed: 0
[0,802] Move x: 605,5 y: 22,4 grabbed: 0
[0,809] Move x: 602,3 y: 11,6 grabbed: 0
[0,816] Move x: 598,8 y: 2,6 
[0,823] Mouse cursor left window: 1on_focus_change: window id: 0x1 focused: 0
[1,460] 
text-input: updating IME focus state, ime_focused: 1 ev->focused: 0
[1,460] text-input: leave event
[1,807] on_focus_change: window id: 0x1 focused: 1
[1,807] 
text-input: updating IME focus state, ime_focused: 0 ev->focused: 1
[1,807] 
text-input: updating cursor position: left=2 top=21 width=7 height=16
[1,807] 
text-input: updating IME focus state, ime_focused: 0 ev->focused: 1
[1,807] text-input: enter event
[1,807] text-input: done event: serial: 11 current_commit_serial: 11
[2,235] Mouse cursor entered window: 1 at 202,617188x9,765625
[2,235] Move x: 202,6 y: 9,8 grabbed: 0
[2,242] Move x: 135,0 y: 22,2 grabbed: 0
[2,249] Move x: 67,3 y: 34,9 grabbed: 0
[2,256] Move x: 2,7 y: 49,0 grabbed: 0
[2,264] Mouse cursor left window: 1Mouse cursor entered window: 1 at 0,000000x257,636719
[2,793] Move x: 0,0 y: 257,6 
[2,800] Move x: 10,1 y: 257,6 grabbed: 0
[2,807] Move x: 18,6 y: 257,6 grabbed: 0
[2,814] Move x: 26,4 y: 257,6 grabbed: 0
[2,821] Move x: 33,1 y: 257,3 grabbed: 0
[2,828] Move x: 38,6 y: 257,0 grabbed: 0
[2,835] Move x: 43,0 y: 256,6 grabbed: 0
[2,842] Move x: 46,7 y: 256,3 grabbed: 0
[2,849] Move x: 49,4 y: 255,9 grabbed: 0
[3,394] Move x: 40,9 y: 258,0 grabbed: 0
[3,401] Move x: 26,9 y: 261,1 grabbed: 0
[3,407] Move x: 2,0 y: 266,9 grabbed: 0
[3,415] Mouse cursor left window: 1Mouse cursor entered window: 1 at 5,214844x289,921875
[4,141] Move x: 5,2 y: 289,9 grabbed: 0
[4,149] Move x: 50,9 y: 284,3 grabbed: 0
[4,155] Move x: 94,8 y: 280,1 grabbed: 0
[4,162] Move x: 133,9 y: 276,9 grabbed: 0
[4,169] Move x: 166,4 y: 274,5 grabbed: 0
[4,176] Move x: 192,3 y: 272,9 grabbed: 0
[4,183] Move x: 212,2 y: 272,0 grabbed: 0
[4,191] Move x: 227,9 y: 271,6 grabbed: 0
[4,198] Move x: 239,4 y: 271,6 grabbed: 0
[4,205] Move x: 246,9 y: 271,6 grabbed: 0
[6,508] Press xkb_keycode: 0x1d clean_sym: Control_L composed_sym: Control_L mods: numlock glfw_key: 57442 (LEFT_CONTROL) xkb_key: 65507 (Control_L)
[6,508] on_key_input: glfw key: 0xe062 native_code: 0xffe3 action: PRESS mods: ctrl+numlock text: '' state: 0 sent encoded key to child: ^[ [ 5 7 4 4 2 ; 1 3 3 u 
[6,729] Press xkb_keycode: 0x36 clean_sym: Shift_R composed_sym: Shift_R mods: ctrl+numlock glfw_key: 57447 (RIGHT_SHIFT) xkb_key: 65506 (Shift_R)
[6,729] on_key_input: glfw key: 0xe067 native_code: 0xffe2 action: PRESS mods: ctrl+shift+numlock text: '' state: 0 [6,729] 
text-input: updating cursor position: left=2 top=69 width=7 height=16
sent encoded key to child: ^[ [ 5 7 4 4 7 ; 1 3 4 u 
[6,730] text-input: done event: serial: 12 current_commit_serial: 12
[7,021] Press xkb_keycode: 0x34 clean_sym: colon composed_sym: slash mods: ctrl+shift+numlock glfw_key: 58 (:) xkb_key: 58 (colon) shifted_key: 47 (/) alternate_key: 46 (.)
[7,021] on_key_input: glfw key: 0x3a native_code: 0x3a action: PRESS mods: ctrl+shift+numlock text: '' state: 0 [7,021] 
text-input: updating cursor position: left=2 top=117 width=7 height=16
sent encoded key to child: ^[ [ 5 8 : 4 7 : 4 6 ; 1 3 4 u 
[7,022] text-input: done event: serial: 13 current_commit_serial: 13
[7,144] Release xkb_keycode: 0x34 clean_sym: colon mods: ctrl+shift+numlock glfw_key: 58 (:) xkb_key: 58 (colon) shifted_key: 47 (/) alternate_key: 46 (.)
[7,144] on_key_input: glfw key: 0x3a native_code: 0x3a action: RELEASE mods: ctrl+shift+numlock text: '' state: 0 [7,144] 
text-input: updating cursor position: left=2 top=181 width=7 height=16
sent encoded key to child: ^[ [ 5 8 : 4 7 : 4 6 ; 1 3 4 : 3 u 
[7,144] text-input: done event: serial: 14 current_commit_serial: 14
[7,248] Release xkb_keycode: 0x36 clean_sym: Shift_R mods: ctrl+shift+numlock glfw_key: 57447 (RIGHT_SHIFT) xkb_key: 65506 (Shift_R)
[7,248] on_key_input: glfw key: 0xe067 native_code: 0xffe2 action: RELEASE mods: ctrl+numlock text: '' state: 0 [7,248] 
text-input: updating cursor position: left=2 top=245 width=7 height=16
sent encoded key to child: ^[ [ 5 7 4 4 7 ; 1 3 3 : 3 u 
[7,248] text-input: done event: serial: 15 current_commit_serial: 15
[7,268] Release xkb_keycode: 0x1d clean_sym: Control_L mods: ctrl+numlock glfw_key: 57442 (LEFT_CONTROL) xkb_key: 65507 (Control_L)
[7,268] on_key_input: glfw key: 0xe062 native_code: 0xffe3 action: RELEASE mods: numlock text: '' state: 0 [7,268] 
text-input: updating cursor position: left=2 top=293 width=7 height=16
sent encoded key to child: ^[ [ 5 7 4 4 2 ; 1 2 9 : 3 u 
[7,269] text-input: done event: serial: 16 current_commit_serial: 16
[8,440] Move x: 243,8 y: 271,9 grabbed: 0
[8,449] Move x: 239,6 y: 271,9 grabbed: 0
[8,454] Move x: 234,2 y: 272,2 grabbed: 0
[8,461] Move x: 228,4 y: 272,2 grabbed: 0
[8,468] Move x: 222,0 y: 272,6 grabbed: 0
[8,476] Move x: 214,5 y: 272,9 grabbed: 0
[8,483] Move x: 207,0 y: 273,6 grabbed: 0
[8,490] Move x: 198,9 y: 274,3 grabbed: 0
[8,497] Move x: 190,4 y: 275,3 grabbed: 0
[8,504] Move x: 182,3 y: 276,3 grabbed: 0
[8,511] Move x: 175,1 y: 277,7 grabbed: 0
[8,518] Move x: 168,7 y: 279,0 grabbed: 0
[8,525] Move x: 162,6 y: 280,0 grabbed: 0
[8,532] Move x: 157,5 y: 281,1 grabbed: 0
[8,541] Move x: 153,1 y: 281,7 grabbed: 0
[8,549] Move x: 149,7 y: 282,4 grabbed: 0
[8,554] Move x: 146,6 y: 282,8 grabbed: 0
[8,561] Move x: 143,9 y: 283,1 grabbed: 0
[8,568] Move x: 141,2 y: 283,4 grabbed: 0
[8,575] Move x: 138,8 y: 283,8 grabbed: 0
[8,582] Move x: 136,4 y: 284,1 grabbed: 0
[8,590] Move x: 133,7 y: 284,4 grabbed: 0
[8,596] Move x: 130,7 y: 284,8 grabbed: 0
[8,604] Move x: 127,6 y: 285,1 grabbed: 0
[8,611] Move x: 123,9 y: 285,5 grabbed: 0
[8,618] Move x: 119,5 y: 286,1 grabbed: 0
[8,625] Move x: 114,7 y: 286,5 grabbed: 0
[8,632] Move x: 109,6 y: 287,2 grabbed: 0
[8,640] Move x: 104,5 y: 287,8 grabbed: 0
[8,646] Move x: 99,4 y: 288,2 grabbed: 0
[8,653] Move x: 94,7 y: 288,9 grabbed: 0
[8,660] Move x: 90,3 y: 289,2 grabbed: 0
[8,668] Move x: 86,2 y: 289,5 grabbed: 0
[8,674] Move x: 82,5 y: 289,9 grabbed: 0
[8,681] Move x: 79,4 y: 290,2 grabbed: 0
[8,690] Move x: 76,4 y: 290,6 grabbed: 0
[8,696] Move x: 74,0 y: 290,9 grabbed: 0
[8,703] Move x: 71,9 y: 291,2 grabbed: 0
[8,710] Move x: 70,2 y: 291,2 grabbed: 0
[8,718] Move x: 68,9 y: 291,2 grabbed: 0
[8,725] Move x: 67,5 y: 291,2 grabbed: 0
[8,732] Move x: 66,5 y: 291,2 grabbed: 0
[8,740] Move x: 65,8 y: 291,2 grabbed: 0
[8,746] Move x: 65,2 y: 291,2 grabbed: 0
[8,753] Move x: 64,5 y: 291,2 grabbed: 0
[8,760] Move x: 64,1 y: 291,2 grabbed: 0
[8,767] Move x: 63,9 y: 291,2 grabbed: 0
[8,774] Move x: 63,5 y: 291,6 grabbed: 0
[8,781] Move x: 63,2 y: 291,6 grabbed: 0
[9,268] Move x: 58,7 y: 293,5 grabbed: 0
[9,275] Move x: 52,6 y: 295,9 grabbed: 0
[9,282] Move x: 44,1 y: 298,9 grabbed: 0
[9,290] Move x: 33,3 y: 302,8 grabbed: 0
[9,296] Move x: 20,6 y: 308,0 grabbed: 0
[9,303] Move x: 7,0 y: 313,8 grabbed: 0
[9,311] Mouse cursor left window: 1[9,771] Press xkb_keycode: 0x1d clean_sym: Control_L composed_sym: Control_L mods: numlock glfw_key: 57442 (LEFT_CONTROL) xkb_key: 65507 (Control_L)
on_key_input: glfw key: 0xe062 native_code: 0xffe3 action: PRESS mods: ctrl+numlock text: '' state: 0 [9,771] 
text-input: updating cursor position: left=2 top=341 width=7 height=16
sent encoded key to child: ^[ [ 5 7 4 4 2 ; 1 3 3 u 
[9,771] text-input: done event: serial: 17 current_commit_serial: 17
[9,937] Press xkb_keycode: 0x2e clean_sym: c composed_sym: c mods: ctrl+numlock glfw_key: 99 (c) xkb_key: 99 (c)
[9,937] on_key_input: glfw key: 0x63 native_code: 0x63 action: PRESS mods: ctrl+numlock text: '' state: 0 [9,937] 
text-input: updating cursor position: left=2 top=389 width=7 height=16
sent encoded key to child: ^[ [ 9 9 ; 1 3 3 u 
[9,938] text-input: done event: serial: 18 current_commit_serial: 18
[9,941] Popped key encoding flags to: 0
[9,941] 
text-input: updating IME focus state, ime_focused: 1 ev->focused: 0
```

And what I already produced on the first comment
```
ctrl+num_lock+LEFT_CONTROL PRESS 
CSI 57442 ; 133 u

shift+ctrl+num_lock+RIGHT_SHIFT PRESS 
CSI 57447 ; 134 u

shift+ctrl+num_lock+: PRESS 
CSI 58 : 47 : 46 ; 134 u
Shifted key: / Alternate key: . 

shift+ctrl+num_lock+: RELEASE 
CSI 58 : 47 : 46 ; 134 : 3 u
Shifted key: / Alternate key: . 

ctrl+num_lock+RIGHT_SHIFT RELEASE 
CSI 57447 ; 133 : 3 u

num_lock+LEFT_CONTROL RELEASE 
CSI 57442 ; 129 : 3 u
```

**Comment by metal3d** [comment]
Are you kidding ? It's now the 4th time I copy paste the logs

**Comment by kovidgoyal** [comment]
No I am not kidding you, I needed both logs from the same run so I can correlate the events, which you finally posted in your previous comment which crossed my previous comment.

**Comment by kovidgoyal** [comment]
Now from the log we see:
```
Press xkb_keycode: 0x34 clean_sym: colon composed_sym: slash mods: ctrl+shift+numlock glfw_key: 58 (:) xkb_key: 58 (colon) shifted_key: 47 (/) alternate_key: 46 (.)
```

this indicates you are pressing ctrl+shift+colon key. Presumably in your layout shift+colon is /.  From the kitten output:

```
shift+ctrl+num_lock+: PRESS 
CSI 58 : 47 : 46 ; 134 u
Shifted key: / Alternate key: . 
```

we see that kitty is correctly encoding the event as ctrl+shift+colon (ignoring the num lock which is irrelevant here) with the shifted key being / as expected.

Now it is upto nvim on how it wants to match this key event. It can match it either as ctrl+shift+: or ctrl+/ or both, more correctly.

This is not in kitty's control, as I said it needs to be reported to neovim. It should settle on a strategy for how to map shifted shortcuts. kitty can only report the pressed keys, how they get mapped to actions in the final application is not in kitty's control.

**Comment by metal3d** [comment]
Here: https://github.com/kovidgoyal/kitty/issues/8046#issuecomment-2473526197 I already gave the logs.

And **THEN**, you asked me a second time to give you logs, being agressive. I regave the logs here https://github.com/kovidgoyal/kitty/issues/8046#issuecomment-2473568477 for the second time

And **THEN** you re-asked me a third time because there were a latency on your emails. This one, I can accept that there were cross comment, but the 2 firsts times, that was not the case. I don't understand why you keep arguing that I didn't give the logs, when if you look at the chronology, I did it long before you accused me of “thinking I know more than you do”.

I'm going to assume that you've misunderstood, or misread, or that you're having a hard day and I've been acting like a punching bag... And I'm not a person known for holding grudges. So we'll move on.

I will try to explain this to NeoVim developers, pointing on this issue to have some logs on the keys I pressed and your recommendations.

PS: The remaining question is "why CTRL+\" (so on AZERTY keyboard: CTRL+8) replaces the shortcut. It's not yourside, I understand.

**Comment by kovidgoyal** [comment]
On Wed, Nov 13, 2024 at 05:37:15AM -0800, Patrice Ferlet wrote:
> Here: https://github.com/kovidgoyal/kitty/issues/8046#issuecomment-2473526197 I already gave the logs.

That is the --debug-input log WITHOUT the kitten output. I wanted BOTH.
Instead you gave me a wev log.

>
> And **THEN**, you asked me a second time to give you logs, being agressive. I regave the logs here https://github.com/kovidgoyal/kitty/issues/8046#issuecomment-2473568477 for the second time

You realize that from the moment I even read your original issue
description, I knew this was not a bug in kitty, and said so in about 5
minutes from you opening this bug report. At that point if you had
simply listened to me and gone and reported the issue to neovim you
would have saved SO MUCH TIME. I have wasted about
an hour of my valuable time simply to convince *you* of that fact. And you
think I am being aggressive! You do realise my open source projects are
used by millions of people and I have to support all of them *on my
own*, for free. Check your expectations at the door.

>
> And **THEN** you re-asked me a third time because there were a latency on your emails. This one, I can accept that there were cross comment, but the 2 firsts times, that was not the case. I don't understand why you keep arguing that I didn't give the logs, when if you look at the chronology, I did it long before you accused me of “thinking I know more than you do”.


I asked for two logs together. You did not provide them. I have
no confidence that the separate logs are for the same session
without that and I have quite enough to do without wasting my time
convincing you of something that is obvious to me without you even
providing proper logs.

>
> I'm going to assume that you've misunderstood, or misread, or that you're having a hard day and I've been acting like a punching bag... And I'm not a person known for holding grudges. So we'll move on.
>
> I will try to explain this to NeoVim developers, pointing on this issue to have some logs on the keys I pressed and your recommendations.
>
> PS: The remaining question is "why CTRL+\" (so on AZERTY keyboard: CTRL+8) replaces the shortcut. It's not yourside, I understand.

The explanation is the same look at the logs and the output carefully
and you will understand it.

**Comment by kovidgoyal** [comment]
> > That is the --debug-input log WITHOUT the kitten output. I wanted BOTH. Instead you gave me a wev log.
> 
> This is on the very first comment

Yes, except you didnt actually say that is what it was so I had no way of knowing that short of reading it and guessing that it is the output I wanted.

**Comment by metal3d** [comment]
The fist comment gave the outputs of the command to log keys, you asked me the second parts, I gave you on the second comment (as soon as I could)

As I explained: only kitty has got the problem, so, being logical, and WITHOUT SAYING You'RE WRONG, I explained that only kitty seems to fails, so probably you missed something (I'm developper, I know that sometimes we missed a possibility)

You've simply decided that I'm an idiot, or that I'm trying to prove you wrong, while I'm giving you the information as best I can.

Because, well, pulling up logs on two windows, knowing that the slightest mouse movement will line up dozens of extra lines, and then coming to reproach me for not giving them to you the way “you” want, and on top of that attacking me... that's a lot.

Add to that the fact that I'm making quite an effort to understand English, a language that isn't my own, to explain what I'm seeing, read the documentation, and try to understand to potentially find out what's going wrong...

This is the first time in over 24 years that a software author has treated me this way. I've always been happy to talk to the devs of major projects, and the people who talk to me about my projects I don't throw away like garbage.

So, once again, I suggest you leave it at that. We've misunderstood each other, but I don't deserve this aggression and since you're sticking to your guns, I'd like it to stop.
